core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageContainer.java

```diff
@@ -144,9 +146,17 @@ public ModuleUpdateStatus getModuleUpdateStatus(String moduleKey) {
     return getComponentByType(IssueStoreReader.class).getServerIssues(moduleKey, filePath);
   }
 
-  public Iterator<ServerIssue> downloadServerIssues(String moduleKey, String filePath) {
-    // TODO
-    return getServerIssues(moduleKey, filePath);
+  public Iterator<ServerIssue> downloadServerIssues(ServerConfiguration serverConfig, String moduleKey, String filePath) {
+    IssueStoreReader issueStoreReader = getComponentByType(IssueStoreReader.class);
+    StorageManager storageManager = getComponentByType(StorageManager.class);
+    IncrementalUpdate updater = IncrementalUpdate.create(storageManager, serverConfig, issueStoreReader);
+
+    try {
+      updater.update(moduleKey, filePath);
+    } catch (Exception e) {
```

Comment:
We talked about throwing a functional exception that IntelliJ-side can catch and log in case of download failures.

Note that IntelliJ-side currently tries to download first and if fails then get. But in this implementation download will never fail, so IntelliJ-side is pointless.

So maybe it will be better to rethrow a functional exception here (and not try to return from store), and update IntelliJ to catch the exception (and fall back to getting from store, as it already does).